src/docs/cookbook-manage/enable-login-as-user/index.md

```diff
@@ -78,179 +78,10 @@ const authenticated = authInfo =>
-  const userId = req.query.user_id;
-
-  if (!userId) {
-    return res.status(400).send('Missing query parameter: user_id.');
-  }
-  if (!ROOT_URL) {
-    return res
-      .status(409)
-      .send('Marketplace canonical root URL is missing.');
-  }
-
-  const state = urlifyBase64(crypto.randomBytes(32).toString('base64'));
-  const codeVerifier = urlifyBase64(
-    crypto.randomBytes(32).toString('base64')
-  );
-  const hash = crypto
-    .createHash('sha256')
-    .update(codeVerifier)
-    .digest('base64');
-  const codeChallenge = urlifyBase64(hash);
-  const authorizeServerUrl = `${CONSOLE_URL}/api/authorize-as`;
-
-  const location = `${authorizeServerUrl}?\
-response_type=code&\
-client_id=${CLIENT_ID}&\
-redirect_uri=${loginAsRedirectUri}&\
-user_id=${userId}&\
-state=${state}&\
-code_challenge=${codeChallenge}&\
-code_challenge_method=S256`;
-
-  const cookieOpts = {
-    maxAge: 1000 * 30, // 30 seconds
-    secure: USING_SSL,
-  };
-
-  res.cookie(stateKey, state, cookieOpts);
-  res.cookie(codeVerifierKey, codeVerifier, cookieOpts);
-  return res.redirect(location);
-});
-
-// Works as the redirect_uri passed in an authorization code request. Receives
-// an authorization code and uses that to log in and redirect to the landing
-// page.
-router.get('/login-as', (req, res) => {
-  const { code, state, error } = req.query;
-  const storedState = req.cookies[stateKey];
-
-  if (state !== storedState) {
-    return res.status(401).send('Invalid state parameter.');
-  }
-
-  if (error) {
-    return res
-      .status(401)
-      .send(`Failed to authorize as a user, error: ${error}.`);
-  }
-
-  const codeVerifier = req.cookies[codeVerifierKey];
-
-  // clear state and code verifier cookies
-  res.clearCookie(stateKey, { secure: USING_SSL });
-  res.clearCookie(codeVerifierKey, { secure: USING_SSL });
-
-  const baseUrl = BASE_URL ? { baseUrl: BASE_URL } : {};
-  const tokenStore = sharetribeSdk.tokenStore.expressCookieStore({
-    clientId: CLIENT_ID,
-    req,
-    res,
-    secure: USING_SSL,
-  });
-
-  const sdk = sharetribeSdk.createInstance({
-    transitVerbose: TRANSIT_VERBOSE,
-    clientId: CLIENT_ID,
-    httpAgent: httpAgent,
-    httpsAgent: httpsAgent,
-    tokenStore,
-    typeHandlers: [
-      {
-        type: sharetribeSdk.types.BigDecimal,
-        customType: Decimal,
-        writer: v => new sharetribeSdk.types.BigDecimal(v.toString()),
-        reader: v => new Decimal(v.value),
-      },
-    ],
-    ...baseUrl,
-  });
-
-  sdk
-    .login({
-      code,
-      redirect_uri: loginAsRedirectUri,
-      code_verifier: codeVerifier,
-    })
-    .then(() => res.redirect('/'))
-    .catch(() =>
-      res.status(401).send('Unable to authenticate as a user')
-    );
-});
-
-module.exports = router;
-```
+add them to the Node server, copy
+[server/apiRouter.js](https://github.com/sharetribe/ftw-daily/blob/master/server/apiRouter.js),
+`server/api/initiate-login-as.js`, and `server/api/login-as.js` files
```

Comment:
I think it would be nice if these two files were also links to FTW-repo

Reply:
I wanted to avoid direct links to files since those are bound to change in the long term. But it doesn't really make sense to have one file linked and two others not. So you have a point. Would it work if we only linked to the `server/` directory and then mentioned the file names that should be changed?

Reply:
@kpuputti if we want to avoid having direct links to files then linking to the directory would be a good solution. I don't have strong opinion on which one we should pick but thought it feels a little weird that one file is linked and other ones are not...